### PR TITLE
Add datalayer sample apps

### PR DESCRIPTION
**Pull request by luizgrp**
#### WHAT

Add datalayer sample apps.

#### WHY

https://github.com/google/horologist/issues/1808

#### HOW

- Add new wear project to folder datalayer/sample/wear.
- Add new phone project to folder datalayer/sample/phone.
- Add new library project to folder sample/shared.
- Move datalayer sample code from auth/sample/shared project to datalayer/sample/shared project.
- Move datalayer sample code from auth/sample/phone project to datalayer/sample/phone project.
- Move datalayer sample code from auth/sample/wear project to datalayer/sample/wear project.

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [N/A] Update metalava's signature text files

### CODE REVIEW

File: datalayer/sample/phone/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            android:theme="@style/Theme.Horologist">
+
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+
+                <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>
+        </activity>
+
+        <service android:name=".grpc.WearCounterDataService" android:exported="true"
```
Comment (luizgrp) [inline datalayer/sample/phone/src/main/AndroidManifest.xml:44]: remove

Comment (luizgrp) [reply datalayer/sample/phone/src/main/AndroidManifest.xml:44]: wrong line, meant to remove from auth/sample/phone

**Review by kul3r4 — APPROVED** [review]

File: datalayer/sample/phone/proguard-rules.pro
Comment (yschimke) [inline datalayer/sample/phone/proguard-rules.pro:1]: We should move this into a library and stop sprinkling it everywhere.

File: datalayer/sample/phone/src/main/java/com/google/android/horologist/datalayer/sample/ui/theme/HorologistTheme.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    onPrimary = Color.White,
+    onSecondary = Color.White,
+    onTertiary = Color.White,
+    onBackground = Color(0xFF1C1B1F),
+    onSurface = Color(0xFF1C1B1F),
+     */
+)
+
+@Composable
+fun HorologistTheme(
```
Comment (yschimke) [inline datalayer/sample/phone/src/main/java/com/google/android/horologist/datalayer/sample/ui/theme/HorologistTheme.kt:57]: Do we care about themes here?

File: datalayer/sample/shared/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    resourcePrefix = "horologist_"
+
+    namespace = "com.google.android.horologist.datalayer.sample.shared"
+}
+
+protobuf {
+    protoc {
+        artifact = "com.google.protobuf:protoc:3.25.0"
```
Comment (yschimke) [inline datalayer/sample/shared/build.gradle.kts:77]: Some existing tech debt, we should move these into the toml file.

**Comment by yschimke** [comment]
Landing, nothing blocking.

**Pull request merged**